dota2/features/lobby.py

```diff
@@ -0,0 +1,221 @@
+from dota2.enums import ESOType, EGCBaseMsg, EDOTAGCMsg
+
+
+class Lobby(object):
+    EVENT_LOBBY_INVITE = 'lobby_invite'
+    """When a lobby invite is received
+    :param message: `CSDOTALobbyInvite <https://github.com/ValvePython/dota2/blob/ca75440adca20d852b9aec3917e4387466848d5b/protobufs/dota_gcmessages_common_match_management.proto#L100>`_
+    :type message: proto message
+    """
+    EVENT_LOBBY_INVITE_END = 'lobby_invite_end'
+    """When a lobby invite is no longer valid
+    :param message: `CSDOTALobbyInvite <https://github.com/ValvePython/dota2/blob/ca75440adca20d852b9aec3917e4387466848d5b/protobufs/dota_gcmessages_common_match_management.proto#L100>`_
+    :type message: proto message
+    """
+    EVENT_LOBBY_NEW = 'lobby_new'
+    """Entered a lobby, either by creating one or accepting an invite
+
+    :param message: `CSODOTALobby <https://github.com/ValvePython/dota2/blob/ca75440adca20d852b9aec3917e4387466848d5b/protobufs/dota_gcmessages_common_match_management.proto#L193>`_
+    :type message: proto message
+    """
+    EVENT_LOBBY_CHANGED = 'lobby_changed'
+    """Anything changes to the lobby state, players, options, broadcasters...
+
+    :param message: `CSODOTALobby <https://github.com/ValvePython/dota2/blob/ca75440adca20d852b9aec3917e4387466848d5b/protobufs/dota_gcmessages_common_match_management.proto#L193>`_
+    :type message: proto message
+    """
+
+    lobby = None
+    lobby_invite = {}
+
+    def __init__(self):
+        super(Lobby, self).__init__()
+        self.socache.on(('new', ESOType.CSODOTALobbyInvite), self.__handle_lobby_invite)
+        self.socache.on(('removed', ESOType.CSODOTALobbyInvite), self.__handle_lobby_invite_removed)
+
+        self.on(EDOTAGCMsg.EMsgGCPracticeLobbyResponse, self.__handle_lobby_new)
```

Comment:
This doesn't seem right. That handler will set `.lobby` with the response message, not CSOLobby.